src/main/java/ch/njol/skript/lang/function/FunctionRegistry.java

```diff
@@ -0,0 +1,588 @@
+package ch.njol.skript.lang.function;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptAPIException;
+import com.google.common.base.Preconditions;
+import org.jetbrains.annotations.ApiStatus.Internal;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.converter.Converters;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A registry for functions.
+ *
+ * @author Efnilite
+ */
+@Internal // for now
+final class FunctionRegistry {
+
+	private FunctionRegistry() {
+		throw new UnsupportedOperationException("Cannot instantiate utility class");
+	}
```

Comment:
This probably shouldn't be based on static stuff, it's a registry so it should be instantiated and used via reference.